docs/reference/modules/discovery/bootstrapping.asciidoc

```diff
@@ -56,6 +56,28 @@ cluster.initial_master_nodes:
   - master-node-hostname
 --------------------------------------------------
 
+[NOTE]
+==================================================
+The node names used in this list must exactly match the `node.name` properties
+of the nodes. If your node names are a fully-qualified domain name such as
+`master-a.example.com` then you must include the fully-qualified domain name in
+the `cluster.initial_master_nodes` list too; conversely if your node names are
+bare hostnames (without the `.example.com` suffix) then you must use bare
+hostnames in the `cluster.initial_master_nodes` list too. If you use a mix of
```

Comment:
> If you use a mix of fully-qualifed and bare hostnames then the cluster will not properly form and
you will see log messages like the following, indicating that the names do not match exactly.

How about: if the node names and the `cluster.initial_master_nodes` list don't much exactly, you are likely to see log message like the following:

Reply:
I want to keep the bit about a "mix of fully-qualifed and bare hostnames" but I added some more general words in eacb7640de0 too.